pd/src/components/shielded_pool.rs

```diff
@@ -271,27 +269,48 @@ impl ShieldedPool {
             },
             source,
         )
-        .await;
+        .await?;
 
         Ok(())
     }
 
     #[instrument(skip(self, source, output_body))]
-    async fn add_note(&mut self, output_body: output::Body, source: NoteSource) {
+    async fn add_note(&mut self, output_body: output::Body, source: NoteSource) -> Result<()> {
```

Comment:
This function shouldn't return a `Result`, because we cannot fail to execute a transaction once we've accepted it with `check_tx_stateful`.

Instead, we need to check that the `BlockTree` is not full in the `check_tx_stateful` method, and then `.expect` (with a reference to that check) that the `BlockTree` is not full in `execute_tx`.

Reply:
The `Component` API doesn't make it clear that this is a problem, because there's no distinction in what *kind* of `Result`s are returned.  Discussion of a potential fix: https://github.com/penumbra-zone/penumbra/issues/632